src/fsharp/service/ServiceLexing.fs

```diff
@@ -696,13 +727,33 @@ type FSharpLineTokenizer(lexbuf: UnicodeLexing.Lexbuf,
             | _ ->
                 // Get the information about the token
                 let (colorClass, charClass, triggerClass) = TokenClassifications.tokenInfo token
+
                 let lexcontFinal =
                     // If we're using token from cache, we don't move forward with lexing
                     if isCached then lexcontInitial else LexerStateEncoding.computeNextLexState token lexcontInitial
-                let tokenTag = tagOfToken token
+
+                let tokenTag =
+                    // Tokenization just reports repeated STRING_TEXT then STRING for each part of an interpolated string
```

Comment:
We did a review of the feature and decided that these new tokens **should** be revealed in the FCS API.  

For example, imagine implementing automatic brace completion, which is currently done based on token information (noirmally incrementally maintained).  We would only want to do brace completion for braces in interpolated strings and not braces in normal strings, e.g. 
```
$""  --> INTERP_STRING_BEGIN_END ""
$"abc"  --> INTERP_STRING_BEGIN_END "abc"
$"abc{x}"  --> INTERP_STRING_BEGIN_PART "abc{"; IDENT "x"; INTERP_STRING_END "}"
$"abc{x}-{y}"  --> INTERP_STRING_BEGIN_PART "abc{"; IDENT "x"; INTERP_STRING_PART "}-{"; IDENT "y"; INTERP_STRING_END "}"
```
Possible rule:
```
   Give automatic brace completion on (INTERP_STRING_BEGIN_PART text)
   and (INTERP_STRING_PART text) where text ends in "{" and "{" has just been typed
```